### DISCUSSION TITLE
Parameters to layout.tsx

### DISCUSSION THREAD

**Discussion by edwinvelez** [vercel/next.js#45023] [Ideas]
With Next.js 13, is it possible to interact directly with `layout.tsx`? If not, I think this would be a nice feature to have. Someone else also has [a similar issue](https://github.com/vercel/next.js/discussions/43871).

DISCLAIMER: I have left similar notes leading up to this discussion on the Beta documentation.

I know I can create [Nested Layouts](https://beta.nextjs.org/docs/routing/pages-and-layouts#nesting-layouts) and [Multiple Root Layouts](https://beta.nextjs.org/docs/routing/defining-routes#example-creating-multiple-root-layouts).

What though if I wanted to selectively show a component defined in `/layout.tsx`? For example, I want `<Navbar />` to show on most pages but not all i.e. anything inheritied from `layout.tsx` but not on `/dashboard/page.tsx` that has its own `layout.tsx` file. I could also have a landing page where I do not want the `<Navbar />` shown.  If `/layout.tsx` were to accept parameters, it could help simplify (or maybe even eliminate) the use of Nested and Multiple Root Layouts. It would definitely clean up lines of code which may become repitive depending on how many `layout.tsx` files exist in the project.

Would this also contribute to reducing page render counts? I admit, I am not 100% fully confident in how it exactly works with Next 13. Going back to the example of `<Navbar />`, when changing between `/dashboard/page.tsx` and `/page.tsx`. This would be two page renders between routes correct for each `layout.tsx`? But if it accepted parameters and `<Navbar />` was conditionally shown in a single `layout.tsx` file, would this help in that count?

**Comment by Asbjoedt** [comment]
I have the same issue right now. I want to pass parameters to layout.tsx to conditionally render a sidebar's nav menu items based on arguments from page.tsx.
